client/lib/fifo/fifo_unix.go

```diff
@@ -0,0 +1,25 @@
+// +build !windows
+package fifo
+
+import (
+	"io"
+	"os"
+	"syscall"
+)
+
+func New(path string) (io.ReadWriteCloser, error) {
+	err := syscall.Mkfifo(path, 0600)
+	if err != nil {
+		return nil, err
+	}
+
+	return Open(path)
+}
+
+func Open(path string) (io.ReadWriteCloser, error) {
```

Comment:
Hm, should we make an OpenReader and OpenWriter API to avoid misuse? `New` could return a Reader as it should be the reader's responsibility to create the fifo as needed.

Reply:
Yeah I like this. The writer should also always be the one to close it I think so `New` and `OpenReader` will just return an `io.Reader` and `OpenWriter` can return a `io.WriteCloser`.

Reply:
Actually I don't think we will ever need an OpenReader. Plus it doesn't work on windows since creating the pipe returns a listener.

Reply:
The logger will read and write into the log file. And we'll need a writer for the driver-side.

The read side will close when the write side closes, so I don't think the reader will need to be a Close method but it wouldn't hurt. Might be useful for tests or future use cases.

The Windows and Unix APIs should match to avoid accidental build errors on the Windows side.